packages/openapi3/test/test-openapi-output.ts

```diff
@@ -378,6 +378,276 @@ describe("openapi3: definitions", () => {
+      }
+      model Beagle extends Dog {
+        purebred: boolean;
+      }                                                                                            
+                                                                                                    
+      @route("/")                                                                                  
+      namespace root {                                                                             
+        op read(): { @body body: Pet };                                                            
+      }                                                                                            
+      `);
+    ok(openApi.components.schemas.Cat, "expected definition named Cat");
+    ok(openApi.components.schemas.Dog, "expected definition named Dog");
+    ok(openApi.components.schemas.Beagle, "expected definition named Beagle");
+    ok(openApi.components.schemas.Pet, "expected definition named Pet");
+    deepStrictEqual(openApi.paths["/"].get.responses["200"].content["application/json"].schema, {
+      $ref: "#/components/schemas/Pet",
+    });
+    deepStrictEqual(openApi.components.schemas.Pet, {
+      type: "object",
+      properties: { name: { type: "string" }, weight: { type: "number", format: "float" } },
+      required: ["name"],
+      discriminator: { propertyName: "kind" },
+      oneOf: [
+        { $ref: "#/components/schemas/Cat" },
+        { $ref: "#/components/schemas/Dog" },
+        { $ref: "#/components/schemas/Beagle" },
+      ],
+    });
+  });
+
+  it("adds mapping entries to the discriminator when appropriate", async () => {
+    const openApi = await openApiFor(`                                                             
+      @discriminator("kind")                                                                       
+      model Pet { }                                                                                
+      model Cat extends Pet {                                                                      
+        kind: "cat" | "feline";                                                                              
+        meow: int32;                                                                               
+      }                                                                                            
+      model Dog extends Pet {                                                                      
+        kind: "dog";                                                                              
+        bark: string;                                                                              
+      }
+      model Lizard extends Pet {                                                                      
+        kind: string;                                                                                                                                                           
```

Comment:
I *think* this one should cause an error diagnostic, for the same reasons as above